TrackletAlgorithm/ProjectionCalculator.h

```diff
@@ -207,7 +207,8 @@ template<
   constexpr int izproj3 = zmean[9];
   const ap_int<14> zproj3 = izproj3;
   projToDisk(zproj3, itinv, rinv, phi0, t, z0, ir_D4, iphi_D4);
-  const bool valid_D4 = ir_D4 >= irmindisk && ir_D4 < irmaxdisk && ((t > itcut) || (t<-itcut));
+  constexpr int iphi_max = (1<<14); //Should have propper constant for 14 
```

Comment:
@aryd I think we may already have a constant for the magic 14. Assuming this cut on `iphi_D4` corresponds to the following cut in the emulation:
https://github.com/cms-L1TK/cmssw/blob/e9a64590c22c0f03f55d3894c1f745acc6d103ab/L1Trigger/TrackFindingTracklet/src/ProjectionCalculator.cc#L297
then I think the 14 we need is already here in Constants.h:
https://github.com/cms-L1TK/firmware-hls/blob/fc59320c951a9eacf22c5244595dcfe4187d03d6/TrackletAlgorithm/Constants.h#L32

Also, if I'm interpreting the cuts correctly, I think this would need to be `(1<<14)-1` to exactly match the behavior of the emulation:
```suggestion
  constexpr int iphi_max = (1<<n_phidisk_)-1;
```

Finally, should this cut not be applied to the other disk projections? I don't see anything in the emulation that would make this specific to D4.